ibm/service/vpc/resource_ibm_is_bare_metal_server.go

```diff
@@ -905,6 +923,88 @@ func resourceIBMISBareMetalServerCreate(context context.Context, d *schema.Resou
 					}
 					inlinenicobj = append(inlinenicobj, nicobj)
 				}
+			} else if interfaceTypeOk, ok := nic[isBareMetalServerNicInterfaceType]; ok {
+				interfaceType = interfaceTypeOk.(string)
+				var nicobj = &vpcv1.BareMetalServerNetworkInterfacePrototypeBareMetalServerNetworkInterfaceByHiperSocketPrototype{}
+				nicobj.InterfaceType = &interfaceType
+
+				subnetintf, _ := nic[isBareMetalServerNicSubnet]
+				subnetintfstr := subnetintf.(string)
+				nicobj.Subnet = &vpcv1.SubnetIdentity{
+					ID: &subnetintfstr,
+				}
+				name, _ := nic[isBareMetalServerNicName]
+				namestr := name.(string)
+				if namestr != "" {
+					nicobj.Name = &namestr
+				}
+
+				enableInfraNAT, ok := nic[isBareMetalServerNicEnableInfraNAT]
+				enableInfraNATbool := enableInfraNAT.(bool)
+				if ok {
+					nicobj.EnableInfrastructureNat = &enableInfraNATbool
+				}
+
+				if primaryIpIntf, ok := nic[isBareMetalServerNicPrimaryIP]; ok && len(primaryIpIntf.([]interface{})) > 0 {
+					// primaryIp := primaryIpIntf.([]interface{})[0].(map[string]interface{})
```

Comment:
Please remove commented code if they are not relevant and will not be used while releasing..